### PR TITLE
[CSM] Add function to set/get the camera mode

### PR DESCRIPTION
**Pull request by bigfoot547**

### CODE REVIEW

**Comment by red-001** [comment]
I think we should implement a separate object for the camera or make it part of the local player api.

**Comment by nerzhul** [comment]
camera is not related directly to player, imagine you control camera and you make machinima ?

File: src/camera.h [hunk 1]
```diff
@@ -160,6 +160,16 @@ class Camera
 		else
 			m_camera_mode = CAMERA_MODE_FIRST;
 	}
+	
+	// Set the current camera mode
+	inline bool setCameraMode(CameraMode cmode) {
```
Comment (nerzhul) [inline src/camera.h]: brace on wrong line

File: src/script/lua_api/l_client.cpp [hunk 2]
```diff
@@ -88,6 +90,50 @@ int ModApiClient::l_get_player_names(lua_State *L)
 	return 1;
 }
 
+// set_camera_mode(mode)
+int ModApiClient::l_set_camera_mode(lua_State *L)
+{
+	CameraMode cmode = static_cast<CameraMode>(lua_tonumber(L, 1));
+	Client *client = getClient(L);
+	if (!client)
+	{
```
Comment (nerzhul) [inline src/script/lua_api/l_client.cpp]: brace on wrong line

File: src/script/lua_api/l_client.cpp [hunk 3]
```diff
@@ -88,6 +90,50 @@ int ModApiClient::l_get_player_names(lua_State *L)
 	return 1;
 }
 
+// set_camera_mode(mode)
+int ModApiClient::l_set_camera_mode(lua_State *L)
+{
+	CameraMode cmode = static_cast<CameraMode>(lua_tonumber(L, 1));
```
Comment (nerzhul) [inline src/script/lua_api/l_client.cpp]: no static_cast here, a lua_tonumber is never a CameraMode

**Comment by nerzhul** [comment]
missing doc

**Comment by bigfoot547** [comment]
@red-001I agree. I think that I will work on a camera class api like the localplayer api tomorrow.

**Comment by bigfoot547** [comment]
Replaced by #5609.

**Pull request closed without merging**